src/resources/js/admin-image-field.js

```diff
@@ -0,0 +1,172 @@
+/* eslint-disable linebreak-style */
+/**
+ * Makes sure we have all the required levels on the Tribe Object.
+ * 
+ * @since TBD
+ * 
+ * @type {PlainObject}
+ */
+tribe.settings = tribe.settings || {};
+tribe.settings.fields = tribe.settings.fields || {};
+
+/**
+ * Configure image field for settings in global Tribe variable.
+ * 
+ * @since TBD
+ * 
+ * @type {PlainObject}
+ */
+tribe.settings.fields.image = {};
+
+/**
+ * Intializes the image field for settings.
+ * 
+ * @since TBD
+ * 
+ * @param {PlainObject} $   jQuery
+ * @param {PlainObject} obj tribe.settings.fields.image
+ */
+( function( $, obj ) {
+	'use strict';
+	var $document = $( document );
+
+	/**
+	 * Store the frame object globally.
+	 * 
+	 * @since TBD
+	 * 
+	 * @type {PlainObject}
+	 */
+	obj.frame = false;
+
+	/**
+	 * Selectors used for configuration and setup
+	 *
+	 * @since TBD
+	 *
+	 * @type {PlainObject}
+	 */
+	obj.selectors = {
+		imageFieldContainer: '.tribe-field-image',
+		addImgLink: '.tec-admin__settings-image-field-btn-add',
+		removeImgLink: '.tec-admin__settings-image-field-btn-remove',
+		imgContainer: '.tec-admin__settings-image-field-image-container',
+		imgIdInput: '.tec-admin__settings-image-field-input',
+	};
+
+	/**
+	 * Method to hide/show elements based on the image input field.
+	 * 
+	 * @since TBD
+	 * 
+	 * @param {jQuery} $fieldParent jQuery object of the field container.
+	 *
+	 * @return {void}
+	 */
+	obj.hideElements = function( $fieldParent ) {
+		const imageIsSet = $fieldParent.find( obj.selectors.imgIdInput ).val() !== '';
+		$fieldParent.find( obj.selectors.addImgLink ).toggleClass( 'hidden', imageIsSet );
+		$fieldParent.find( obj.selectors.removeImgLink ).toggleClass( 'hidden', !imageIsSet );
+		$fieldParent.find( obj.selectors.imgContainer ).toggleClass( 'hidden', !imageIsSet );
+	};
+
+	/**
+	 * Method to handle when image is selected from WP Media feature.
+	 * 
+	 * @since TBD
+	 * 
+	 * @param {jQuery} $fieldParent jQuery object of the field container.
+	 *
+	 * @return {void}
+	 */
+	obj.onImageSelect = function( $fieldParent ) {
+		const attachment = obj.frame.state().get('selection').first().toJSON(),
+			$imgContainer = $fieldParent.find( obj.selectors.imgContainer );
+		if ( $imgContainer.find('img').length > 0 ){
```

Comment:
Have we changed the JS linting rules? I see some spacing incongruencies within the JS file.
```suggestion
		if ( $imgContainer.find('img').length > 0 ) {
```